### DISCUSSION TITLE
Support for Public Routes

### DISCUSSION THREAD

**Discussion by Be-P** [directus/directus#19581] [Feature Requests]
### Summary

It would be really usefull to have the possibility of defining Public Routes, especially in the case of implementing custom registration flows.

I've seen the Login and Accept Invite Routes already have the feature of defining public Routes. It would be great to define Public Modules and/or Public Child Routes.

### Basic Example

```javascript
export default defineModule({
	id: 'custom',
	name: 'Custom',
	icon: 'box',
	meta: {
		public: true,
	},
	routes: [
		{
			path: 'public',
			component: ModuleComponent,
			meta: {
				public: true
			}
		},
		{
			path: 'private',
			component: ModuleComponent,
			meta: {
				public: false
			}
		},
	],
});

```

### Motivation

An example use case is the implementation of custom user registration flow.

The documentation says that you can hook on the user registration with pwd and token, while it is actually not true because overriding with `USER_INVITE_URL_ALLOW_LIST` you only get back the token. 

To provide the user with custom registration flow, eg requiring more info during the registration, you would need to redirect the user to a publicly accessible Route and allow the final user fill the custom registration form

### Detailed Design

I think this section doesn't need too much content, the Basic Example section and the overall context should be enough to provide a "Detailed Design" for this feature request

### Must Have:
- Possibility of defining Public Routes

### Should Have:
- Possibility of defining both Public and Private Routes in the same Module

### Drawbacks

There aren't many drawbacks.

### Alternatives

I don't think there are alternatives, a mature Directus Implementation I think should provide Public Routes

### Adoption Strategy

I don't think much content is needed in this section

**Comment by Be-P** [comment]
I'm preparing a prototype PR, I'll need some help to refine the code. LMK if there is some hype on this I'll spend more time on it
Edit. link to the PR https://github.com/directus/directus/pull/19587

**Reply by rijkvanzanten** [reply]
Lets hold any implementation work until this feature request has made it's way through [the feature request process](https://docs.directus.io/contributing/feature-request-process.html#request-a-feature) 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
